stdlib/fnmatch.pyi

```diff
@@ -1,9 +1,10 @@
+from _typeshed import GenericPath
 from collections.abc import Iterable
 from typing import AnyStr
 
 __all__ = ["filter", "fnmatch", "fnmatchcase", "translate"]
 
-def fnmatch(name: AnyStr, pat: AnyStr) -> bool: ...
-def fnmatchcase(name: AnyStr, pat: AnyStr) -> bool: ...
-def filter(names: Iterable[AnyStr], pat: AnyStr) -> list[AnyStr]: ...
+def fnmatch(name: GenericPath, pat: GenericPath) -> bool: ...
+def fnmatchcase(name: GenericPath, pat: GenericPath) -> bool: ...
+def filter(names: Iterable[AnyStr], pat: GenericPath) -> list[AnyStr]: ...
```

Comment:
`names` has [an optimization](https://github.com/python/cpython/blob/eff2042fac6dc7d6e2f8aeec5a64a3a44a1a9648/Lib/fnmatch.py#L53-L54) which prevents `PathLike` from working on posix at least (probably from before PathLike was a thing), so I left it.